components/button/floating-buttons.js

```diff
@@ -270,6 +276,27 @@ class FloatingButtons extends RtlMixin(LitElement) {
 		});
 	}
 
+	_scrollIfFloatObsuringFocus() {
+
+		if (!this._floating) return;
+
+		const currentFocusedItem = getComposedActiveElement();
+		if (isComposedAncestor(this, currentFocusedItem)) return
+
+		const { y: focusedY, height: focusedHeight } = currentFocusedItem.getBoundingClientRect();
+		const { y: floatingY, height: floatingHeight } = this.shadowRoot.querySelector('.d2l-floating-buttons-container').getBoundingClientRect();
+		if (focusedY === 0 || floatingY === 0) return;
+
+		const isObscuring = (floatingY - focusedY) < Math.max(MINIMUM_TARGET_SIZE, focusedHeight);
```

Comment:
I switched this around to `Math.max` from `Math.min` -- @awikkerink does that make sense? This ensures that the entire height of the focused element isn't being obscured. Otherwise if say a button that's `42px` height was showing at least `24px`, it wouldn't scroll.

I'm actually tempted to remove the `MINIMUM_TARGET_SIZE` bit altogether since anything that's focusable should already be at least `24px` high.

Reply:
The reason I added this logic was if there was a focusable element (usually with tabIndex=-1) that takes up the entire height of the page, clicking it would result in the page being scrolled to the bottom.

Reply:
Hmm... I wonder if that was caused by using `document.activeElement` instead of the composed active element? We could put an upper bound on it maybe... like if the focusable thing is taller than `100px`.

Reply:
![Screenshot 2023-08-30 143217](https://github.com/BrightspaceUI/core/assets/7862298/67508c83-5eba-4ecb-ab39-96ae1e005474)

Reply:
For reference, the page is `/d2l/le/lo/6609/outcomes-gradebook/settings`.

Reply:
> I'm actually tempted to remove the MINIMUM_TARGET_SIZE bit altogether since anything that's focusable should already be at least 24px high.

I'm not checking if the element is at least 24px high, but only scrolling if enough of the element is obscured to avoid scrolling down the entire page for a large element.

Reply:
Ok I'm convinced, I'll flip this back.

Reply:
I'm fine with increasing the height from 24px to 100px or something. I just didn't want the whole page to scroll